src/pumipic_adjacency.hpp

```diff
@@ -1141,5 +1116,97 @@ bool search_mesh_2d(o::Mesh& mesh, // (in) mesh
   return found;
 }
 
+OMEGA_H_DEVICE o::LO search_mesh_2d(const o::Read<o::I8> side_is_exposed,
```

Comment:
Should the name be changed to make it clearer that this is for a single particle?

Reply:
At the time, I was hesitate to introduce new function name, so just kept the same.

Reply:
I agree about changing the name. Adding `_pt` at the end is good enough to me. 

Using the same name is a good way to confuse me in the future when working on the search functions.

Reply:
I agree it's better to use a different name to avoid future confusion.